### ISSUE TITLE
CRUDing subset of resources

### ISSUE THREAD

**Issue by jmeas** [jmeas/resourceful-redux#174]
Labels have two purposes:

1. tracking every request, even requests that aren't initially linked to resource ID(s)
2. maintaining ordered lists of subsets of a resource

For instance, you might need to display two lists of books on your page. Reading them with different labels gives you those two lists.

Labels clearly solve item 1 above, but for item 2, there are situations where you might need a different abstraction. For instance, what happens when you want to create an item?

Since labels store ID references, updates to the resource are automatically synced. And deletes from the main list will remove the ID from the label. But they don't support creation and appending to the list.

@marlonpp was dealing with this situation the other day. I suggested that maybe he didn't need to use labels, and could just use the full list of resources. This solution works when you don't have more than one list.

---

He suggested a sort of plugin system, and I'm thinking that that _may_ be best for this. Perhaps some way to keep track of lists of resources:

```
{
  resources: {},
  labels: {},
  meta: {},
  lists: {}
}
```

`lists` would be a list of resource IDs that would be updated as you CRUD resources. You could specify a `list` param to your action to keep track.

Or...maybe I should look into how/if Ember Data handles this.

---

I'll likely start this out as a plugin, and we'll see how it goes from there.

**Comment by jmeas** [comment]
Pretty confident that the plan described by #175 / #176 will solve this, so I'm going to close this out for now. If you think there's a better way, just leave a comment here or in one of those issues and we'll keep the conversation going!

**Comment by jmeas** [comment]
Leaving this here: https://www.apollographql.com/docs/react/features/caching.html#updating-the-cache-after-a-mutation

That describes how Apollo solves this problem. For posterity:

```js
const text = 'Hello, world!';

client.mutate({
  mutation: TodoCreateMutation,
  variables: {
    text,
  },
  update: (proxy, { data: { createTodo } }) => {
    // Read the data from our cache for this query.
    const data = proxy.readQuery({ query: TodoAppQuery });

    // Add our todo from the mutation to the end.
    data.todos.push(createTodo);

    // Write our data back to the cache.
    proxy.writeQuery({ query: TodoAppQuery, data });
  },
});
```

What this does is:

1. Runs the mutation query
2. When the query is done, the user imperatively updates a separate read query cache to include the todo

tl;dr, this is very manual, which is a good thing
